bin/crew

```diff
@@ -578,36 +578,58 @@ def download
     case File.basename(filename)
     # Sources that download with curl
     when /\.zip$/i, /\.(tar(\.(gz|bz2|xz|lz))?|tgz|tbz|txz)$/i, /\.deb$/i
-      # Recall file from cache if requeseted
+      # Recall file from cache if requested
       if CREW_CACHE_ENABLED
+        puts "Looking for archive in cache".orange if @opt_verbose
         cachefile = CREW_CACHE_DIR + filename
-        if File.file?(cachefile)
+        if ! File.exist?(cachefile)
+          puts 'Cannot find cached archive. 😔 Will download.'.lightred
+          cachefile = ''
+        else
+          puts "Archive file exists in cache".lightgreen if @opt_verbose
           if Digest::SHA256.hexdigest( File.read(cachefile) ) == sha256sum then
-            FileUtils.cp cachefile, CREW_BREW_DIR, verbose: @fileutils_verbose
-            puts "Archive found in cache".lightgreen
-            return {source: source, filename: filename}
+            begin
+              # Hard link cached file if possible.
+              FileUtils.ln cachefile, CREW_BREW_DIR, verbose: @fileutils_verbose
```

Comment:
Why a hard link and why call it cachefile when you are linking to a directory?

Reply:
CREW_BREW_DIR is a dir, but `ln filename dir` should make a hard linked file in `dir`, no?

Reply:
I suppose that is true if the directory exists.